eloquent-javascript-exercises/chap4/chap4.js

```diff
@@ -0,0 +1,107 @@
+//write program range function to take an optional third argument that indicates the “step” value used when building the array. If no step is given, the elements go up by increments of one, corresponding to the old behavior.
+function range(start, end, step = 1) {
+    let numbers = [];
+    if (step > 0) {
+        for (let i = start; i <= end; i += step) {
+            numbers.push(i);
+        }
+    }
+    else {
+        for (let i = start; i >= end; i += step) {
+            numbers.push(i);
+        }
+    }
+    return numbers;
+}
+
+
+
+//write a sum function that takes an array of numbers and returns the sum of these numbers.
+
+
+function sum(array) {
+    return array.reduce((total, num) => total + num);
```

Comment:
this looks incorrect
 array.reduce((total, num) => total + num, 0);
aggregator is missing, is default value 0 here?